source/common/interruptDispatcher.s

```diff
@@ -67,46 +74,65 @@ BEGIN_ASM_FUNC IntrMain
     orr     r3, r3, r1
     str     r3, [r0]
 #endif
-endsetflags:
 
+    // find the highest set IRQ bit
+    // input:
+    // r1 = interrupt mask
+    // r2 = irq table address
+    // output:
+    // r1 = target interrupt mask
+    // r2 = target irq table address
+    // r0, r3 can be used freely
 findIRQ:
-
-    ldr     r0, [r2, #4]
-    cmp     r0, #0
+#ifdef ARM9
+    clz     r0, r1
+    eor     r0, r0, #31
+#else
```

Comment:
Can you add a comment here explaining a bit what the code below is doing? Is it just the equivalent of `clz`?

Reply:
The `clz`/`eor` and ARM7 code are equivalent; it's just that the `clz` instruction is ARM9-exclusive.